app/interactors/destroy_map.rb

```diff
@@ -0,0 +1,12 @@
+class DestroyMap
+  include Interactor
+
+  def call
+    if context.campaign.current_map == context.map
+      ChangeCurrentMap.call(campaign: context.campaign, map: nil)
+    end
```

Comment:
I see what you mean. It doesn't feel great, especially with the context branching. We're not catching a failure of `ChangeCurrentMap` (although the exception handling model in interactors is something I know that I don't understand so it might automatically trigger a rollback)

Perhaps we can change `SetCurrentMap` to look for `context.current_map` and do nothing if it's not present? Or we could prepare `context.campaign_params` as an interactor when we're working with maps, update the map then check what has changed in the broadcast interactors to see whether they need updating?

Reply:
Ok, I've come back at this and I think the "proper" way to handle this is to ensure that each interactor only does one thing by extracting that additional behavior into another Interactor and moving that into the Organizer.

I've done this in 214215f.

This also ensures that if unsetting the current map fails, the interaction fails and the destroy is not attempted, which wasn't the case in the previous code.

Is it a lot of code? Yes. But it is "correct" and well-tested? Yes. I think the awkward naming (suggestions welcome) is not great, but assuming we want to continue to push on interactors (which for the time being I'm still OK with) I think this is what we get.

I'd also note again that there is an open PR for adding conditionals to Organizers, which would allow us to extract the conditional out of the Interactor and name, and up into a method/check on the Organizer. But it's been sitting for some time.

Reply:
I think this looks like a good approach. I've left some suggestions about the name.